test/e2e/iam/azure_iam_roles.jsonc

```diff
@@ -0,0 +1,177 @@
+/* Azure role definitions for E2E tests.
+
+   Set of permissions required by the TriggerMesh end-to-end test suite to run
+   tests that manipulate Azure resources.
+*/
+[
+  {
+    "roleName": "TriggerMesh E2E test suite",
+    "description": "Grants permissions required by the TriggerMesh E2E test suite.",
+    "permissions": [
+      {
+        "actions": [
+          "Microsoft.EventHub/namespaces/read",
+          "Microsoft.EventHub/namespaces/write",
+          "Microsoft.Devices/iothubs/iothubkeys/listkeys/action",
+          "Microsoft.Devices/iothubs/read",
+          "Microsoft.Devices/iothubs/write",
+          "Microsoft.Resources/subscriptions/resourceGroups/delete",
+          "Microsoft.Resources/subscriptions/resourceGroups/write",
+          "Microsoft.ServiceBus/namespaces/authorizationrules/listKeys/action",
+          "Microsoft.ServiceBus/namespaces/read",
+          "Microsoft.ServiceBus/namespaces/write",
+          "Microsoft.Storage/storageAccounts/blobServices/containers/write",
+          "Microsoft.Storage/storageAccounts/blobServices/write",
+          "Microsoft.Storage/storageAccounts/listKeys/action",
+          "Microsoft.Storage/storageAccounts/write"
+        ],
+        "dataActions": [
+          "Microsoft.EventHub/namespaces/messages/send/action",
+          "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/delete",
+          "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write"
+        ],
+        "notActions": [],
+        "notDataActions": []
+      }
+    ],
+    "assignableScopes": [
+      "/subscriptions/52e4a8f0-1968-40a4-b38b-f19fac1e89ce"
+    ]
+  },
+  {
+    "roleName": "Activity Logs event source",
+    "description": "Role suitable for use with the TriggerMesh event source for Activity Logs.",
+    "permissions": [
+      {
+        "actions": [
+          "Microsoft.EventHub/namespaces/authorizationRules/listkeys/action",
```

Comment:
Is this still required as I had removed the listkeys on the EventHub test.

Reply:
It is. This is the `Activity Logs event source` role, it contains everything required to run the **source**, not the E2E test orchestration.

Azure requires this permission while interacting with the Activity Logs "diagnostic settings" API to be able to perform the necessary wiring on the Event Hubs side.
In API calls, you pass the ID of the rule that should be used, and Activity Logs uses the aforementioned permission to read it:

https://github.com/triggermesh/triggermesh/blob/06a157cadcd1098d7d5daa3a22a94fa090f3d6bb/pkg/sources/reconciler/azureactivitylogssource/diagsettings.go#L163-L164

The requirement is also documented here:

https://github.com/triggermesh/triggermesh/blob/06a157cadcd1098d7d5daa3a22a94fa090f3d6bb/pkg/sources/reconciler/azureactivitylogssource/diagsettings.go#L62-L66

Reply:
Gotcha. Was under the impression this was more for the test framework itself.

Reply:
I tried to separate them to make it clear what is used by the test logic, and what is used by each TriggerMesh components, but it's not easy because you can't [group permissions like in AWS](https://github.com/triggermesh/triggermesh/blob/06a157cadcd1098d7d5daa3a22a94fa090f3d6bb/test/e2e/aws_iam_policy.jsonc#L10-L12), so overlaps aren't obvious.